source/concepts/applications.rst

```diff
@@ -0,0 +1,358 @@
+* ``rotation`` is a quaternion describing the rotation of the user's game
+  space in the server space.
+
+A client may not follow the server suggestion and should not be assumed to do
+so. If the key is absent from the shared state, the client may locate itself in
+the server space as it chooses.
+
+.. warning::
+
+   A client has no way of knowing if the user origin emanates from a ligitimate
+   source (i.e. the server or a trusted client). Therefore, a client could use
+   this feature to move users without their conscent. This could cause
+   discomfort if not used responsibly.
+
+As a summary, the user origin is specified as follow in the shared state:
+
+.. code::
+
+   user-origin.<PLAYER_ID>: {
+     position,
+     rotation,
+   }
+
+.. _play-space-description:
+
+Play space
+~~~~~~~~~~
+
+A client can share the shape of its play space with the others. A play space,
+or play area, is the area the user can safely reach. This is mostly relevant
+for VR clients which have to define such a safe space.
+
+The play area is defined as four points, each as a vector of 3 XYZ values, in
+server space, that form a quadrilateral. The play area is defined as a
+Struct in the shared state under the key ``playarea.<PLAYER_ID>``. The points
+are defined under the keys ``A``, ``B``, ``C``, and ``D``.
+
+.. code::
+
+   playarea.<PLAYER_ID>: {
+      A,
+      B,
+      C,
+      D,
+    }
+
+If they are available, a client can choose to represent them as they choose.
+
+.. note::
+
+   We assume that the points defining the play area are on the floor (Y=0).
+   However, nothing forces a client to send them a such.
+
+.. _radial-orient:
+
+Radial orient
+~~~~~~~~~~~~~
+
+A server can, optionally, implement the radial orient feature as a command on
+the :ref:`command service <command-service>`. The radial orient command places
+all the avatars on a circle around the origin of the server space by
+setting a :ref:`user origin <user-origin-description>` for each avatar.
+
+The command is named ``multiuser/radially-orient-origins``. It takes a
+``radius`` argument that is the distance, in meters, between the generated
+centers and the center of the server space. The default radius is 1 meter.
+The command does not return anything. This leads to the following signature:
+
+.. code::
+
+   multiuser/radially-orient-origins(radius = 1.0) -> None
+
+Let a set of players :math:`P = \{P_0, P_1, ... P_{N - 1}\}`, :math:`N` the number of
+players, and :math:`r` the radius given in argument. Then the center's position
+:math:`C_i` for avatar :math:`i` is computed using polar coordinates converted
+to Cartesian. Each avatar is assigned an angle :math:`\theta_i`:
+
+.. math::
+
+  \theta_i = \frac{i \times 2 \pi}{N}
+
+Then the positions is:
+
+.. math::
+
+  \begin{align}
+  C_i &= \begin{bmatrix}
+    r\cos{\theta_i}\\
+    0\\
+    r\sin{\theta_i}\\
+  \end{bmatrix}
+  \end{align}
+
+The rotation :math:`R_i` is expressed as a quaternion and is defined as:
```

Comment:
@Ragzouken do you remember how the rotation works in the radial orient command? I found the formula, but I do not see why it is this way. I guess part of it is there to rotate the space by 90°somehow?

Reply:
The point is to rotate so that the z axis (forward) points inwards to the center. My diagram shows why a 90° turn makes sense. However, it looks like the signs of angles arent consistent and I'm not sure what to make of that.

![IMG_20240506_123358339~2](https://github.com/IRL2/nanover/assets/527158/0c44c212-5b27-41f6-81e3-b2ff9cf2144e)